File: ui/src/parser/profile/cookieandkate.ts
import { Profile } from "./profile";

export class CookieAndKate implements Profile {
  domain = 'cookieandkate.com';  
  
  directions($: CheerioStatic): string[] {
    return $('div.tasty-recipe-instructions li').map((_i, el) => $(el).text()).get();
  }
  ingredients($: CheerioStatic): string[] {
    return $('div.tasty-recipe-ingredients li').map((_i, el) => $(el).text()).get();
  }
  title($: CheerioStatic): string {
    return $('h1.entry-title').text();
  }
}

File: service/RecipeMd.Domain/RecipeSiteProfiles/CookieAndKate.cs
using RecipeMd.Domain.Interfaces;

namespace RecipeMd.Domain.RecipeSiteProfiles
{
    public class CookieAndKateProfile : IRecipeSiteProfile
    {
        public string Domain => "cookieandkate.com";

        public string TitleSelector => "h1.entry-title";

        public string IngredientSelector => "div.tasty-recipe-ingredients li";

        public string DirectionSelector => "div.tasty-recipe-instructions li";
    }
}
